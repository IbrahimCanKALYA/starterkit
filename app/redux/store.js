/* @flow */
import { applyMiddleware, createStore } from 'redux';
import createSagaMiddleware from 'redux-saga';
import logger from 'redux-logger';
import { persistStore, persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';

import sagas from 'sagas';
import { startIntroductionScreen, startTabBasedApp } from '../index';
import reducers from './reducerCombiner';

const sagaMiddleware = createSagaMiddleware();

const persistConfig = {
  key: 'root',
  storage,
};

const persistedReducer = persistReducer(persistConfig, reducers);

const store = createStore(persistedReducer, applyMiddleware(sagaMiddleware, logger));

sagaMiddleware.run(sagas);

function onRehydrate() {
  // Getting past state from phone storage...
  const currentState = store.getState();

  // $FlowFixMe
  console.disableYellowBox = true; // eslint-disable-line

  if (currentState.auth.login) {
    startTabBasedApp();
  } else {
    startIntroductionScreen();
  }
}

persistStore(store, {}, onRehydrate);

export default store;
