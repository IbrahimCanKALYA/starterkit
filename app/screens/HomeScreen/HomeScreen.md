# HomeScreen

## Description

- absent !

## Funtions

- absent !

## Props

- absent !

## NavigateTo

- absent !