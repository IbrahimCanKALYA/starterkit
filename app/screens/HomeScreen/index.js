/* @flow */
import React, { Component } from 'react';
import { View, Text } from 'react-native';
import { connect } from 'react-redux';
import styles from './styles';

type Props = any;

class HomeScreen extends Component<Props> {
  render() {
    return (
      <View style={styles.container}>
        <Text>HomeScreen</Text>
      </View>
    );
  }
}

const mapStateToProps = null;

export default connect(mapStateToProps)(HomeScreen);
