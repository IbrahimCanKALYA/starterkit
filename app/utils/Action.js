/* @flow */
const REQUEST = 'REQUEST';
const SUCCESS = 'SUCCESS';
const FAILURE = 'FAILURE';

function createRequestTypes(base: string) {
  return [REQUEST, SUCCESS, FAILURE].reduce((acc, type) => {
    acc[type] = `${base}_${type}`;
    return acc;
  }, {});
}

function action(type: string, payload: any = {}) {
  return { type, ...payload };
}

module.exports = {
  REQUEST,
  SUCCESS,
  FAILURE,
  createRequestTypes,
  action,
};
