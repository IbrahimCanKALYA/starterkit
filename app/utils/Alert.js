/* @flow */
import { Alert } from 'react-native';

export default {
  // Shows an alert without cancel button...
  showInfo: (
    title: string = 'info',
    message: string,
    ok: Function = () => {
      // $FlowFixMe
      console.log('OK Pressed'); // eslint-disable-line
    },
  ) => {
    Alert.alert(title, message, [{ text: 'Tamam', onPress: ok }], { cancelable: false });
  },
  // Show alert with cancel button for take a confirmation from user...
  confirm: (
    title: string = 'info',
    message: string,
    ok: Function = () => {
      // $FlowFixMe
      console.log('OK Pressed'); // eslint-disable-line
    },
    cancel: Function = () => {
      'CANCEL Pressed';
    },
  ) => {
    Alert.alert(
      title,
      message,
      [{ text: 'Ä°ptal', onPress: cancel }, { text: 'Tamam', onPress: ok }],
      { cancelable: false },
    );
  },
};
