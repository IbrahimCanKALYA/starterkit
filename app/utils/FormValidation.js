/* @flow */

// Validates an email...
const validateEmail = (email: string): boolean => {
  const re = /^(([^<>()\]\\.,;:\s@"]+(\.[^<>()\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
  return re.test(email);
};

// Validates that text has minimum 3 character...
const validateText = (text: string): boolean => !(text.length < 3);

// Validates that password has minimum 8 character...
const validatePasswordLength = (password: string): boolean => !(password.length < 8);

// Validates that password has a special character...
const validatePasswordHaveSpecialChar = (password: string): boolean => {
  const re = /^[a-zA-Z0-9 ]*$/;
  return !re.test(password);
};

// Validates that text is not empty...
const isEmpty = (text: string): boolean => !(text.length === 0);

// Validates that identity number has 11 number...
const validateIdentityNumberLength = (number: string): boolean => !(number.length !== 11);

// Validates that identity characters are numeric...
const validateIdentityNumberNumeric = (number: string): boolean => {
  const re = /^\d+$/;
  return re.test(number);
};

module.exports = {
  validateEmail,
  validateText,
  validatePasswordLength,
  validatePasswordHaveSpecialChar,
  isEmpty,
  validateIdentityNumberLength,
  validateIdentityNumberNumeric,
};
