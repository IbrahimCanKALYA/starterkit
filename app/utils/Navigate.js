/* @flow */
import { Navigation } from 'react-native-navigation';

// Reset the app to a new layout...
const setRoot = (layout: any): void => {
  Navigation.setRoot(layout);
};

// Set default options to all screens. Useful for declaring a consistent style across the app...
const setDefaultOptions = (options: any): void => {
  Navigation.setDefaultOptions(options);
};

// Change a component's navigation options...
const mergeOptions = (componentId: string, options: any): void => {
  Navigation.mergeOptions(componentId, options);
};

// Show a screen as a modal...
const showModal = (screenName: string, passProps: any): void => {
  Navigation.showModal({
    component: {
      name: screenName,
      passProps: {
        ...passProps,
      },
    },
  });
};

// Dismiss a modal by componentId. The dismissed modal can be anywhere in the stack...
const dismissModal = (componentId: string): void => {
  Navigation.dismissModal(componentId);
};

// Dismiss all Modals...
const dismissAllModals = (): void => {
  Navigation.dismissModal();
};

// Push a new layout into this screen's navigation stack...
const push = (componentId: string, screenName: string, passProps: any): void => {
  Navigation.push(componentId, {
    component: {
      name: screenName,
      passProps: {
        ...passProps,
      },
    },
  });
};

// Pop a component from the stack, regardless of it's position...
const pop = (componentId: string, params: any): void => {
  Navigation.pop(componentId, params);
};

// Pop the stack to a given component...
const popTo = (componentId: string): void => {
  Navigation.popTo(componentId);
};

// Pop the component's stack to root...
const popToRoot = (componentId: string): void => {
  Navigation.popToRoot(componentId);
};

// Sets new root component to stack...
const setStackRoot = (componentId: string, layout: any): void => {
  Navigation.setStackRoot(componentId, layout);
};

// Show overlay on top of the entire app...
const showOverlay = (layout: string): void => {
  Navigation.showOverlay(layout);
};

// dismiss overlay by componentId...
const dismissOverlay = (componentId: string): void => {
  Navigation.dismissOverlay(componentId);
};

// Resolves arguments passed on launch...
const getLaunchArgs = () => Navigation.getLaunchArgs();

// Obtain the events registry instance...
const events = () => Navigation.events();

// Constants coming from native...
const constants = () => Navigation.constants();

module.exports = {
  setRoot,
  setDefaultOptions,
  mergeOptions,
  showModal,
  dismissModal,
  dismissAllModals,
  push,
  pop,
  popTo,
  popToRoot,
  setStackRoot,
  showOverlay,
  dismissOverlay,
  getLaunchArgs,
  events,
  constants,
};
