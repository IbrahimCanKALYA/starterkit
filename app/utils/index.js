/* @flow */
import Navigate from './Navigate';
import FormValidation from './FormValidation';
import Alert from './Alert';
import Redux from './Redux';
import Action from './Action';

export {
  Navigate, FormValidation, Alert, Redux, Action,
};
