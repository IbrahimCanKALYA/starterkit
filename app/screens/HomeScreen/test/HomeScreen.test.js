import 'react-native';
import renderer from 'react-test-renderer';
import { Redux } from 'utils';
import store from 'store';
import HomeScreen from '../index';

test('HomeScreen renders correctly', () => {
  const tree = renderer.create(Redux.reduxStoreWrapper(HomeScreen, store)).toJSON();
  expect(tree).toMatchSnapshot();
});
