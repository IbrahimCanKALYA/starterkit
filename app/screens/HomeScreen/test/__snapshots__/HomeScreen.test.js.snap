// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen renders correctly 1`] = `null`;
